source/user/dft.rst

```diff
@@ -504,28 +154,82 @@ nonnegative kinetic energy density
+  >>> Ts = np.einsum('g,g->', weights, ts)
+
+or - as an alternative - the same may be achieved in the following way:
 
-    from pyscf.dft import gen_grid, numint
-    orbo = mf.mo_coeff[:,mf.mo_occ>0]
-    grids = gen_grid.Grids(mol)
-    grids.build(with_non0tab=True)
-    weights = grids.weights
-    ao1 = numint.eval_ao(mol, grids.coords, deriv=1, non0tab=grids.non0tab)
-    ts = 0.5 * numpy.einsum('xgp,pi,xgq,qi->g', ao1[1:], orbo, ao1[1:], orbo)
-    Ts = numpy.einsum('g,g->', weights, ts)
+  >>> Ts_ao = mol_hf.intor('int1e_kin')
+  >>> Ts_analyt = np.einsum('ui,uv,vi->', occ_orbs, Ts_ao, occ_orbs)
 
-    Ts_ao = mol.intor("int1e_kin")
-    Ts_anal = np.einsum("ui,uv,vi->", orbo, Ts_ao, orbo)
-    print(asb(Ts - Ts_anal))
+.. _user_dft_disp_corr:
 
 Dispersion corrections
 ======================
-Grimme's "D3" dispersion correction :cite:`DFTD3` can be added with
-an interface to the external library `libdftd3 <https://github.com/cuanto/libdftd3>`_.
-See :mod:`dftd3`.
+
+Two main ways exist for adding dispersion (van der Waals) corrections to KS-DFT calculations. One is to augment mean-field results by Grimme's D3 corrections :cite:`DFTD3`, which can be added through an interface to the external library `libdftd3 <https://github.com/cuanto/libdftd3>`_, cf. `dftd3/00-hf_with_dftd3.py <https://github.com/pyscf/pyscf/blob/master/examples/dftd3/00-hf_with_dftd3.py>`_:
+
+  >>> from pyscf import dftd3
+  >>> mf_hf_d3 = dftd3.dftd3(dft.RKS(mol_hf))
+  >>> mf_hf_d3.kernel()
+  
+Alternatively, non-local correlation may be added through the VV10 functional :cite:`vydrov_voorhis_vv10_functional_jcp_2010`, cf. `dft/33-nlc_functionals.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/33-nlc_functionals.py>`_:
+
+  >>> mf_hf.xc = 'wb97m_v'
+  >>> mf_hf.nlc = 'vv10'
+  >>> mf_hf.grids.atom_grid = {'H': (99, 590), 'F': (99, 590)}
+  >>> mf_hf.grids.prune = None
+  >>> mf_hf.nlcgrids.atom_grid = {'H': (50, 194), 'F': (50, 194)}
+  >>> mf_hf.nlcgrids.prune = dft.gen_grid.sg1_prune
+  >>> mf_hf.kernel()
+  
+It's important to keep in mind that the evaluation of the VV10 functional involves a double grid integration, so differences in grid size can make an enormous difference in time.
+
+.. _user_dft_pbc:
+
+Periodic Boundary Conditions
+============================
+
+Besides finite-sized systems, PySCF further supports KS-DFT calculations with PBCs for performing solid-state calculations. The APIs for molecular and crystalline input parsing has deliberately been made to align to the greatest extent possible, and an all-electron KS-DFT calculation at the :math:`\Gamma`-point using density fitting (recommended) and a second-order SCF algorithm, requires input on par with a standard KS-DFT calculation for a chemical Hamiltonian (cf. `pbc/11-gamma_point_all_electron_scf.py <https://github.com/pyscf/pyscf/blob/master/examples/pbc/11-gamma_point_all_electron_scf.py>`_):
```

Comment:
```suggestion
Besides finite-sized systems, PySCF further supports KS-DFT calculations with PBCs for performing solid-state calculations. The APIs for molecular and crystalline input parsing has deliberately been made to align to the greatest extent possible, and an all-electron KS-DFT calculation at the :math:`\Gamma`-point using density fitting (recommended) and a second-order SCF algorithm, requires input on par with a standard molecular KS-DFT calculation (cf. `pbc/11-gamma_point_all_electron_scf.py <https://github.com/pyscf/pyscf/blob/master/examples/pbc/11-gamma_point_all_electron_scf.py>`_):
```

"chemical Hamiltonian" is an ambiguous term; I associate it with the approach of Istvan Mayer from the late 1980s and 1990s, see e.g. https://onlinelibrary.wiley.com/doi/abs/10.1002/qua.560230203. I don't think it is available in PySCF. I would just omit the

Reply:
OK